CONTRIBUTING.md

```diff
@@ -1,9 +1,197 @@
+
+Next, we will be notified of the pull request and will read it over. We will try to give an initial response quickly, and then do a longer in-depth review, at which point 
+you will probably need to respond to our comments, making changes as appropriate. We'll then respond again, and proceed in an iterative fashion until everyone is happy with the proposed 
+changes. 
+
+Once your changes are integrated, you will be added as a GitHub contributor and as one of the authors of the package. Thank you for being part of `nemos`!
+
+### Style Guide
+
+The next section will talk about the style of your code and specific requirements for certain feature development in `nemos`. 
+
+- Longer, descriptive names are preferred (e.g., x is not an appropriate name for a variable), especially for anything user-facing, such as methods, attributes, or arguments.
+- Any public method or function must have a complete type-annotated docstring (see below for details). Hidden ones do not need to have complete docstrings, but they probably should.
+
+### Testing 
+
+To run all tests, run `pytest` from within the main `nemos` repository. This may take a while as there are many tests, broken into several categories. 
+There are several options for how to run a subset of tests:
+- Run tests from one file: `pytest tests/test_glm.py`
+- Run a specific test within a specific module: `pytests tests/test_glm.py::test_func`
+- Another example specifying a test method via the command line: `pytest tests/test_glm.py::GLMClass::test_func`
+
+#### Adding tests
+
+New tests can be added in any of the existing `tests/test_*.py` scripts. Tests should be functions, contained within classes. The class contains a bunch of related tests
+(e.g., regularizers, bases), and each test should ideally be a unit test, only testing one thing. The classes should be named `TestSomething`, while test functions should be named 
+`test_something` in snakecase.
+
+If you're adding a substantial bunch of tests that are separate from the existing ones, you can create a new test script. Its name must begin with `test_`, 
+it must have an `.py` extension, and it must be contained within the `tests` directory. Assuming you do that, our github actions will automatically find it and 
+add it to the tests-to-run.
+
+> **_NOTE:_** If you have many variants on a test you wish to run, you should make use of pytest's `parameterize` mark. See the official
+> documentation [here](https://docs.pytest.org/en/stable/how-to/parametrize.html).
+
+> **_NOTE_** If you are using an object that gets used in multiple tests (such as a model with certain data, regularizer, or solver), you should use pytest's `fixtures` to avoid having to 
+> load or instantiate the object multiple times. Look at our `conftest.py` to see already available fixtures for your tests. See the official documentation 
+> [here](https://docs.pytest.org/en/stable/how-to/fixtures.html).
```

Comment:
my recollection is that mixing fixtures and parametrize is slightly more complicated, and reuqires the [`indirect` keyword](https://docs.pytest.org/en/7.1.x/example/parametrize.html#deferring-the-setup-of-parametrized-resources). it doesn't look like you use it nemos, so maybe not worth mentioning here (I use it a lot in plenoptic)

Reply:
no, you don't, I just wanted to point out you should tell people how to do so explicitly if you want people to do it.